drivers/vcluster/vcluster.go

```diff
@@ -0,0 +1,97 @@
+package vcluster
+
+import (
+	"fmt"
+	"github.com/portworx/torpedo/pkg/log"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+var (
+	UpdatedClusterContext string
+	CurrentClusterContext string
+	ContextChange         = false
+)
+
+// This method switches kube context between host and any vcluster
+func SwitchKubeContext(target string) error {
+	cmd := exec.Command("kubectl", "config", "get-contexts", "-o", "name")
+	out, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("Failed to fetch kube contexts: %v", err)
+	}
+	contexts := strings.Split(string(out), "\n")
+	var desiredContext string
+	if target == "host" {
+		for _, ctx := range contexts {
+			if ctx == "kubernetes-admin@cluster.local" {
+				desiredContext = ctx
+				break
+			}
+		}
+	} else {
+		prefix := fmt.Sprintf("vcluster_%s_", target)
+		for _, ctx := range contexts {
+			if strings.HasPrefix(ctx, prefix) {
+				desiredContext = ctx
+				break
+			}
+		}
+	}
+	if desiredContext == "" {
+		return fmt.Errorf("Context for %s not found", target)
+	}
+	log.Infof("Desired Context is : %v", desiredContext)
+	cmd = exec.Command("kubectl", "config", "use-context", desiredContext)
```

Comment:
Same here, can we use k8s API instead?

Reply:
Same answer as above. I don't think we can use api endpoint here for our particular use case. Please let me know if you are able to explore a way out.